lib/ProductOpener/Products.pm

```diff
@@ -1163,22 +1180,53 @@ sub store_product ($user_id, $product_ref, $comment) {
 		}
 	) if $log->is_debug();
 
-	# In case we need to move a product from OFF to OBF etc.
-	# the "new_server" value will be set to off, obf etc.
-	# we first move the existing files (product and images)
-	# and then store the product with a comment.
+	my $delete_from_previous_products_collection = 0;
 
 	# if we have a "server" value (e.g. from an import),
 	# we save the product on the corresponding server but we don't need to move an existing product
+	if (defined $product_ref->{server}) {
+		my $new_server = $product_ref->{server};
+		# Update the product_type from the server
+		if (defined $options{flavors_product_types}{$new_server}) {
+			my $errors_ref = {};
+			change_product_type($product_ref, $options{flavors_product_types}{$new_server}, $errors_ref);
```

Comment:
In case of error, shan't we die ?

Reply:
Diying is a bit harsh, store_product() is called from batch scripts that update / import tons of products, we don't want it to die if there's one issue with one product. I will log an error.

Reply:
I think scripts should call store_product with a try / catch !
Or maybe we could add a parameter to say we wan't a fail-safe call.